### ISSUE TITLE
[REQUEST]: Link to a remote git repository for Save/Load

### ISSUE THREAD

**Issue by kolbyjack** [compiler-explorer/compiler-explorer#3517]
### Is your feature request related to a problem? Please describe

Not applicable

### Describe the solution you'd like

In the Save/Load dialog, it would be nice to have a new tab where you can enter credentials to link to a remote git repository (or use read-only anonymous access), where saving would commit a new revision there, and loading would present a tree of files to select.  Just using the head of the default branch would be great, but being able to browse branches/tags/commits would be even better.

### Describe alternatives you've considered

Not applicable

### Additional context

Not applicable

**Comment by RubenRBS** [comment]
Hi! Thanks for taking the time to request this feature. While it does sound like it would be really useful to connect to a git repo directly from the site (At least for locally running versions!), I don't know how feasible it would be to implement currently.

So while I would love for this feature to exist, don't expect to see this implemented in the near future :/

**Comment by kolbyjack** [comment]
Yeah, after making the request, I realized I was effectively asking to have a js git client.  Maybe if it was restricted to read-only https github/gitlab repos to start with, and then could be expanded to support write access using the respective platform https apis?  Just read access would be an amazing addition, I think.

**Comment by matsjla** [comment]
Maybe something for login support? https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/Roadmap.md#2021-goals

cc @mattgodbolt
